packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -219,6 +230,29 @@ export function dispatchEvent(
     clearIfContinuousEvent(domEventName, nativeEvent);
   }
 
+  if (
+    enableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay &&
+    enableSelectiveHydration &&
```

Comment:
In a follow up, you can remove the enableSelectiveHydration flag. It's always on where it matters and we're committed to some form of it.

Reply:
#22502